ci/functional/test_main_prep_node.sh

```diff
@@ -2,33 +2,226 @@
+                } 2>&1)"
+            mail_message+="${nl}${ib_message}${nl}"
+            echo "$ib_message"
+        else
+            echo "OK: Interface $iface is up."
         fi
-        if ! ifconfig "$iface" | grep "inet "; then
+        if ! ip addr show "$iface" | grep "inet "; then
+            echo "Fail: $ib_message"
+            result=1
             echo "Failed to bring up interface $iface on $HOSTNAME. " \
-            "Please file a CORCI ticket."
-            exit 1
+            "Please file a SRE ticket."
         fi
+        if [ -e "/sys/class/net/$iface/device/numa_node" ]; then
+            set -x
+            cat "/sys/class/net/$iface/device/numa_node"
+        fi
+        set -x
+    else
+        ib_message="Failed to bring up interface $iface on $HOSTNAME. "
+        mail_message+="${nl}${ib_message}${nl}"
+        echo "$ib_message"
+        result=1
+        echo "Please file a SRE ticket."
     fi
 done
 
-if ! grep /mnt/share /proc/mounts; then
+# having -x just makes the console log harder to read.
+set +x
+if [ -e /sys/class/net/ib1 ]; then
+    # now check for pmem & NVMe drives when ib1 is present.
+    # ipmctl show -dimm should show an even number of drives, all healthy
+    dimm_count=0
+    while IFS= read -r line; do
+        if [[ "$line" != *"| Healthy "* ]]; then continue; fi
+        ((dimm_count=dimm_count+1))
+    done < <(ipmctl show -dimm)
+    if [ $dimm_count -eq 0 ] || [ $((dimm_count%2)) -ne 0 ]; then
+       dimm_message="FAIL: Wrong number $dimm_count healthy PMEM DIMMs seen."
+       mail_message+="$nl$dimm_message$nl$(ipmctl show -dimm)$nl"
+    else
+       echo "OK: Found $dimm_count PMEM DIMMs."
+    fi
+    # Should have 2 regions 0x0000 and 0x0001, type AppDirect
+    dimm_rcount=0
+    while IFS= read -r line; do
+        if [[ "$line" != *"| AppDirect"*"| Healthy"* ]]; then continue; fi
+        ((dimm_rcount=dimm_rcount+1))
+    done < <(ipmctl show -region)
+
+    if [ $dimm_rcount -ne 2 ]; then
+       nvme_message="FAIL: Found $dimm_rcount of DIMM PMEM regions, need 2."
+       mail_message+="$nl$nvme_message$nl$(ipmctl show -region)$nl"
+       result=1
+    else
+       echo "OK: Found $dimm_rcount DIMM PMEM regions."
+    fi
+
+    # While this gets more data than needed, it is the same search that
+    # DAOS tests do and records it in the console log.
+    nvme_devices="$(lspci -vmm -D | grep -E '^(Slot|Class|Device|NUMANode):' |
+                  grep -E 'Class:\s+Non-Volatile memory controller' -B 1 -A 2)"
+    nvme_count=0
+    while IFS= read -r line; do
+        if [[ "$line" != *"Class:"*"Non-Volatile memory controller"* ]];then
+            continue
+        fi
+        nvme_count=$((nvme_count+1))
+    done < <(printf %s "$nvme_devices")
+
+    if [ $((nvme_count%2)) -ne 0 ]; then
+       nvme_message="Fail: Odd number ($nvme_count) of NVMe devices seen."
+       mail_message+="$nl$nvme_message$nl$nvmd_devices$nl"
```

Comment:
(lint) nvmd_devices is referenced but not assigned (did you mean 'nvme_devices'?). [SC2154]